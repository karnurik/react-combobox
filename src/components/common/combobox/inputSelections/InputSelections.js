import React, { Component } from 'react';
import './inputSelections.css';

class InputSelections extends Component {
    constructor(props) {
        super(props);
        this.state = {
            slectedElements: []
        }

    }

    /* 
      * @description - Lifecycle method, Using to create array of the selected options
      * @param na 
      * @return na
    */
    componentDidMount () {
       // this.refs.arrowRef.innerHTML = "&#9660;";
       const slectedElements = [];
       this.props.data.forEach(elem => {
           if(this.props.selectedOptions && this.props.selectedOptions.includes(elem.id)){
               slectedElements.push(elem.name)
           }
       });
       this.setState({slectedElements: slectedElements});
    }

    render() {
        const placeholder = this.props.placeholder ? this.props.placeholder : 'Select' ;
        
        return (
            <div className="drop-down"  onClick={this.props.handleInputFocus}>
                <input
                    type='text'
                    className='form-input'
                    placeholder={placeholder}
                    defaultValue= {this.state.slectedElements.join(', ')}
                    />
                    <i className='down-arrow' />
            </div>
        );
    }
}

export default InputSelections;