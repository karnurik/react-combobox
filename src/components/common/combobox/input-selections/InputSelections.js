import React, { Component } from 'react';
import './inputSelections.css';

class InputSelections extends Component {
    constructor(props) {
        super(props);
        this.state = {
            slectedElements: []
        }

    }

    /* 
      * @description - Lifecycle method, Using to create array of the selected options
      * @param na 
      * @return na
    */
    componentDidMount () {
       const slectedElements = [];
       this.props.data.forEach(elem => {
           if(this.props.selectedOptions && this.props.selectedOptions.includes(elem.id)){
               slectedElements.push(elem.name)
           }
       });
       this.setState({slectedElements: slectedElements});
    }

    render() {
        const placeholder = this.props.placeholder ? this.props.placeholder : 'Select' ;
        
        return (
            <>
                <input
                    type='text'
                    className='form-input'
                    placeholder={placeholder}
                    defaultValue= {this.state.slectedElements.join(', ')}
                    onFocus={this.props.handleInputFocus}
                    />
            </>
        );
    }
}

export default InputSelections;