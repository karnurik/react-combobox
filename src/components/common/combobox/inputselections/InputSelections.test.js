import React from 'react';
import InputSelections from './InputSelections';
import renderer from 'react-test-renderer';


const data = [
  { id: '1', name: 'Candy Eat' },
  { id: '2', name: 'Onions Pick' },
  { id: '3', name: 'Fanny Pick' },
  { id: '4', name: 'Click o Wisp'}
];

const slectedElements = ['Candy Eat', 'Onions Pick'];
const placeholder = 'Select Games';

test('Component renders successfully', () => {
  const component = renderer.create(
    <InputSelections
        data={data}
        slectedElements={slectedElements}
        placeholder = {placeholder}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});