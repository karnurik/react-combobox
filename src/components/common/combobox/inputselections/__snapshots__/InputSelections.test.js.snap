// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders successfully 1`] = `
<input
  className="form-input"
  defaultValue=""
  placeholder="Select Games"
  type="text"
/>
`;
